### DISCUSSION TITLE
Error: hash or wait is required at waitForTransaction

### DISCUSSION THREAD

**Discussion by williamlmao** [wevm/wagmi#525] [Q&A]
I am trying to show the user the state of a transaction. Processing, successful, or failed. 

I believe I have followed the documentation to a tee (see my code below), but I am getting this error:

https://wagmi.sh/docs/hooks/useWaitForTransaction#wait-optional

<img width="592" alt="Screen Shot 2022-05-30 at 5 14 20 PM" src="https://user-images.githubusercontent.com/55903529/171059766-60fcbaa0-87f2-4400-aa2f-ca2ffa12ae20.png">

You can see in the console logs that `contractWrite.data?.wait` is not undefined, but I'm still getting this error.  Any help would be greatly appreciated!

```
    const contractWrite = useContractWrite(
    {
      addressOrName: process.env.NEXT_PUBLIC_CONTRACT_ADDRESS,
      contractInterface: abi.abi,
    },
    'purchase',
    {
      args: [
        1,
        {
          value: currentPhase.pricePerToken,
          gasLimit: 148000 + 1 * 2000,
        },
      ],
    }
  )
  console.log('contractWrite.data?.wait', contractWrite.data?.wait)

  const waitForTransaction = useWaitForTransaction({
    wait: contractWrite.data?.wait,
  })

  console.log('waitForTransaction', waitForTransaction)
```

**Comment by YannikSood** [comment]
I am running into a similar issue. I have my write:

```
const write = useContractWrite(
      {
        addressOrName: Address,
        contractInterface: Interface,
      },
      'entry',
      { 
        args: [firstAnswer.trim().toLowerCase(), secondAnswer.trim().toLowerCase()],
        overrides: { 
          value: ethers.utils.parseEther("0.01") 
       }
      }
    )
```

I have my wait:

```
const waitForTransaction = useWaitForTransaction({
       wait: write!.data?.wait
      onSettled(data, error) {
        console.log("Settled", { data })
      }
    });
```

I want to call them in handleSubmit:

```
async function handleSubmit() {
      if (firstAnswer.length == 0 || secondAnswer.length == 0) {
        setInputError(true);
      }
      else {
        setInputError(false);

        const txn = write;
      }
    }
```

nothing happens lol. if i change write to `const { write } = useContractWrite` it is now callable, but I am unable to pass the write data to the useWaitForTransaction hook. Any ideas on how to fix?
